src/reconcile/rec_col.c

```diff
@@ -1653,12 +1675,40 @@ __wt_rec_col_var(
     }
 
     /* If we were tracking a record, write it. */
-    if (rle != 0)
+    if (rle != 0) {
+        if (!last.deleted)
+            wrote_real_values = true;
         WT_ERR(
           __rec_col_var_helper(session, r, salvage, last.value, &last.tw, rle, last.deleted, NULL));
+    }
+
+    /*
+     * If we have not generated any real values but only deleted-value cells, bail out and call the
+     * page empty instead. (Note that this should always be exactly one deleted-value cell, because
+     * of the RLE handling, so we can't have split.) This will create a namespace gap like those
+     * produced by truncate.
+     *
+     * Skip this step if:
+     *     - We're in salvage, to avoid any odd interactions with the way salvage reconstructs the
+     *       namespace.
+     *     - There were invisible updates, because then the page isn't really empty. Also, at least
+     *       for now if we try to restore updates to an empty page col_modify will trip on its
+     *       shoelaces.
+     */
+    if (!wrote_real_values && salvage == NULL && r->leave_dirty == false) {
```

Comment:
Could you add a statistic in this case as well? Again, I think it would be hard to identify this case in a debugger, but can imagine it would be interesting to know that the code had been exercised.

Reply:
Good point again. Added in [c265cda](https://github.com/wiredtiger/wiredtiger/pull/8129/commits/c265cda33a2ef72d4a6f23d411ced096da3f9877).